src/tests/simulation/tests/integration_tests/scale/cordon.rs

```diff
@@ -0,0 +1,419 @@
+    let (fragments, workers) = get_schedule(info.clone());
+    let cordoned_pus_ids = pu_ids_on_cordoned_nodes(&workers);
+    let mut actor_ids_on_cordoned = HashSet::<u32>::new();
+    for frag in fragments {
+        for actor in frag.actor_list {
+            let pu_id = actor.parallel_units_id;
+            if cordoned_pus_ids.contains(&pu_id) {
+                actor_ids_on_cordoned.insert(actor.actor_id);
+            }
+        }
+    }
+    actor_ids_on_cordoned
+}
+
+/// create cluster, run query, cordon node, run other query.
+/// Cordoned node should NOT contain actors from other query
+async fn cordoned_nodes_do_not_get_new_actors(
+    create: &str,
+    select: &str,
+    drop: &str,
+    number_of_nodes: usize,
+) -> Result<()> {
+    tracing_subscriber::fmt()
+        .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+        .init();
+
+    // setup cluster and calc expected result
+    let sleep_sec = 10;
+    let mut cluster =
+        NexmarkCluster::new(Configuration::for_scale(), 6, Some(THROUGHPUT * 20), false).await?;
+    cluster.run(create).await?;
+    sleep(Duration::from_secs(sleep_sec)).await;
+    let expected = cluster.run(select).await?;
+    // do not drop the query
+
+    // cordon random nodes
+    let cordoned_nodes = cluster.cordon_random_workers(number_of_nodes).await?;
+
+    let mut log_msg = "Cordoned the following nodes:\n".to_string();
+    for rand_node in &cordoned_nodes {
+        log_msg = format!("{}{:?}\n", log_msg, rand_node);
+    }
+    tracing::info!(log_msg);
+
+    // check which actors are on cordoned nodes
+    let info = cluster.get_cluster_info().await?;
+    let actors_on_cordoned1 = actor_ids_on_cordoned_nodes(info);
+
+    let dummy_create = "create table t (dummy date, v varchar);";
+    let dummy_mv = "create materialized view mv as select v from t;";
+    let dummy_insert = "insert into t values ('2023-01-01', '1z');";
+    let dummy_select = "select * from mv;";
+    let dummy_drop = "drop materialized view mv; drop table t;";
+    for sql in vec![dummy_create, dummy_mv, dummy_insert] {
+        cluster.run(sql).await?;
+        sleep(Duration::from_secs(sleep_sec)).await;
+    }
+    assert_eq!(cluster.run(dummy_select).await?, "1z");
+
+    // No actors from other query on cordoned nodes
+    let info2 = cluster.get_cluster_info().await?;
+    let actors_on_cordoned2 = actor_ids_on_cordoned_nodes(info2);
+
+    // We allow that an actor moves from a cordoned node to a non-cordoned node
+    // We allow that an actor moves from a cordoned node to another cordoned node
+    // We disallow that an actor moves from a non-cordoned node to a cordoned node
+    assert!(actors_on_cordoned1.is_superset(&actors_on_cordoned2));
+
+    // compare results of original query
+    cluster.run(dummy_drop).await?;
+    sleep(Duration::from_secs(sleep_sec)).await;
+    cluster.run(drop).await?;
+    sleep(Duration::from_secs(sleep_sec)).await;
+    cluster.run(create).await?;
```

Comment:
Do we have some test case where node is cordoned, and `create` fails to run?

Can we add a test for the error behavior in that case?

e.g.
```rs
let result = cluster.run(create).await?;
assert_eq!(result, <some error>);
```

Reply:
Oh nevermind, just saw it below.